buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -1282,6 +1283,27 @@ class BeamModulePlugin implements Plugin<Project> {
         options.errorprone.disableWarningsInGeneratedCode = true
         options.errorprone.excludedPaths = '(.*/)?(build/generated-src|build/generated.*avro-java|build/generated)/.*'
 
+        // Error Prone requires some packages to be exported/opened for Java 17.
+        // This logic handles builds running on JDK 17, notusing -Djava17Home.
+        // The -J prefix is not needed if forkOptions.javaHome is unset,
+        // see http://github.com/gradle/gradle/issues/22747
+        if (JavaVersion.VERSION_1_8.compareTo(JavaVersion.current()) < 0
```

Comment:
Is this actually specific to JDK 17? If so, we should probably do exact equality, if not the comment should be updated.